### PR TITLE
faster download-rg.sh that runs w/o hitting the GitHub API

### PR DESCRIPTION
**Pull request by sqs**
The download-rg.sh script downloads prebuilt ripgrep binaries. Previously, it hit the GitHub API for a list of assets associated with a release. This took ~1-2s. Now, it hardcodes the URL for the assets based on the desired binaries, which should be fairly stable (and if the URL format changes, it will error, not fail silently).

### CODE REVIEW

File: vscode/package.json [hunk 1]
```diff
@@ -10,9 +10,8 @@
   "scripts": {
     "dev": "pnpm run build:dev && CODY_FOCUS_ON_STARTUP=1 NODE_ENV=development code --extensionDevelopmentPath=\"$INIT_CWD/vscode\" --disable-extension=sourcegraph.cody-ai --disable-extension=github.copilot --disable-extension=github.copilot-nightly --inspect-extensions=9333 . --goto ./src/logged-rerank.ts:16:5",
     "generate:completions": "ts-node-transpile-only ./src/testutils/cli/run-code-completions-on-dataset.ts",
-    "build": "tsc --build && pnpm run esbuild --minify && vite build --mode production",
-    "build:dev": "tsc --build && concurrently \"pnpm run esbuild --sourcemap\" \"vite build --mode development\"",
-    "prepare": "scripts/download-rg.sh",
+    "build": "scripts/download-rg.sh && tsc --build && pnpm run esbuild --minify && vite build --mode production",
```
Comment (abeatrix) [inline vscode/package.json:13]:
```suggestion
    "build": "tsc --build && pnpm run esbuild --minify && vite build --mode production",
```
If we are running the `download-rg.sh` script in the prepublish step, which will be called [every time the extension is packaged](https://code.visualstudio.com/api/working-with-extensions/publishing-extension#prepublish-step), does that mean we don't need to rerun the script in the build step?

Comment (abeatrix) [inline vscode/package.json:13]:
```suggestion
    "build": "tsc --build && pnpm run esbuild --minify && vite build --mode production",
```
If we are running the `download-rg.sh` script in the prepublish step, which will be called [every time the extension is packaged](https://code.visualstudio.com/api/working-with-extensions/publishing-extension#prepublish-step), does that mean we don't need to rerun the script in the build step?

File: vscode/package.json [hunk 2]
```diff
@@ -10,9 +10,8 @@
...
+    "build:dev": "scripts/download-rg.sh && tsc --build && concurrently \"pnpm run esbuild --sourcemap\" \"vite build --mode development\"",
```
Comment (abeatrix) [inline vscode/package.json:14]:
```suggestion
    "build:dev": "tsc --build && concurrently \"pnpm run esbuild --sourcemap\" \"vite build --mode development\"",
```

File: vscode/package.json (hunk 1 above)
Comment (sqs) [reply vscode/package.json:13]: I added it here for local dev. It's a footgun in local dev if you don't have the `rg` binaries installed. Now that it's super quick to run (it doesn't even hit the GitHub API again if you've already downloaded the binaries, so it takes like 5ms), I think it's fine to run it every invocation here. That also helps if we bump the required `rg` version when you are on local dev.

File: vscode/package.json (hunk 2 above)
Comment (sqs) [reply vscode/package.json:15]: same reply as https://github.com/sourcegraph/cody/pull/68#discussion_r1262166903

**Review by abeatrix — APPROVED** [review]
Lgtm 😊

**Pull request merged**